probeye/inference/scipy/solver.py

```diff
@@ -195,6 +195,13 @@ def loglike(self, theta: np.ndarray) -> float:
         ll
             The evaluated log-likelihood function for the given theta-vector.
         """
+        
+        # check whether the values of the latent parameters are within the specified domain.
+        # if they are outside of the domain, break calculating the log-likelihood function
+        # and return instead of it minus infinity.
+        if self.problem.check_parameters_domain(theta) == False:
+            return -np.inf
```

Comment:
A more pythonic way to write this condition would be:
```suggestion
        if not self.problem.check_parameters_domain(theta):
            return -np.inf
```